cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java

```diff
@@ -412,4 +388,37 @@ List<URL> getBaseURLs(VOSURI target, URI securityMethod, String scheme) {
         return baseURLs;
     }
 
+    protected static final File findFile(String fname) throws MissingResourceException {
+        File ret = new File(DEFAULT_CONFIG_DIR, fname);
+        if (!ret.exists()) {
+            ret = FileUtil.getFileFromResource(fname, CavernURLGenerator.class);
+        }
+        return ret;
+    }
+
+
+    public static VOSURI getURIFromPath(String path) throws URISyntaxException {
+
+        log.debug("getURIFromPath for " + path);
+        LocalServiceURI localServiceURI = new LocalServiceURI();
+        VOSURI baseURI = localServiceURI.getVOSBase();
+        log.debug("baseURI for cavern deployment: " + baseURI.toString());
+
+        int firstSlashIndex = path.indexOf("/");
+        String pathStr = path.substring(firstSlashIndex + 1);
+        log.debug("path: " + pathStr);
+        String targetURIStr = baseURI.toString() + "/" + pathStr;
+        log.debug("target URI for validating token: " + targetURIStr);
+
+        URI targetURI = new URI(targetURIStr);
+        log.debug("targetURI for system: " + targetURI.toString());
+        VOSURI targetVOSURI = new VOSURI(targetURI);
```

Comment:
There is a constructor for VOSURI that takes a String.  That should remove some of the complication in this method.